tests/shipment_test.go

```diff
@@ -1,221 +1,130 @@
-		)
-		if assert.NotEmpty(shipment.CustomsInfo.CustomsItems) {
-			assert.Equal(
-				customsItem.Description,
-				shipment.CustomsInfo.CustomsItems[0].Description,
-			)
-			assert.Equal(
-				customsItem.HSTariffNumber,
-				shipment.CustomsInfo.CustomsItems[0].HSTariffNumber,
-			)
-			assert.Equal(
-				customsItem.Value, shipment.CustomsInfo.CustomsItems[0].Value,
-			)
-		}
-	}
+	shipmentsList := shipments.Shipments
 
-	require.NotEmpty(shipment.Rates)
-	var rate *easypost.Rate
-	for _, r := range shipment.Rates {
-		if !strings.EqualFold(r.Carrier, "USPS") &&
-			!strings.EqualFold(r.Carrier, "ups") {
-			continue
-		}
-		if !strings.EqualFold(r.Service, "priorityMAILInternational") {
-			continue
-		}
-		if rate == nil {
-			rate = r
-		} else {
-			x, _ := strconv.ParseFloat(r.Rate, 64)
-			y, _ := strconv.ParseFloat(rate.Rate, 64)
-			if x < y {
-				rate = r
-			}
-		}
+	assert.LessOrEqual(len(shipmentsList), c.fixture.pageSize())
+	assert.NotNil(shipments.HasMore)
+	for _, shipment := range shipmentsList {
+		assert.Equal(reflect.TypeOf(&easypost.Shipment{}), reflect.TypeOf(shipment))
 	}
-	require.NotNil(rate)
-	shipment, err = client.BuyShipment(shipment.ID, rate, "100.00")
-	require.NoError(err)
-
-	assert.NotEmpty(shipment.TrackingCode)
-	assert.Equal("100.00", shipment.Insurance)
-	if assert.NotNil(shipment.PostageLabel) {
-		assert.Contains(
-			shipment.PostageLabel.LabelURL,
-			"https://easypost-files.s3-us-west-2.amazonaws.com",
-		)
+}
+
+func (c *ClientTests) TestShipmentBuy() {
+	client := c.TestClient()
+	assert := c.Assert()
+
+	shipment, _ := client.CreateShipment(c.fixture.FullShipment())
+
+	lowestRate, _ := client.LowestRate(shipment)
+
+	boughtShipment, _ := client.BuyShipment(shipment.ID, &lowestRate, "")
+
+	assert.NotNil(boughtShipment.PostageLabel)
+}
+
+func (c *ClientTests) TestShipmentRegenerateRates() {
+	client := c.TestClient()
+	assert := c.Assert()
+
+	shipment, _ := client.CreateShipment(c.fixture.FullShipment())
+
+	rates, _ := client.RerateShipment(shipment.ID)
+
+	assert.Equal(reflect.TypeOf([]*easypost.Rate{}), reflect.TypeOf(rates))
+	for _, rate := range rates {
+		assert.Equal(reflect.TypeOf(&easypost.Rate{}), reflect.TypeOf(rate))
 	}
 }
 
-func (c *ClientTests) TestShipmentList() {
+func (c *ClientTests) TestShipmentConvertLabel() {
 	client := c.TestClient()
-	assert, require := c.Assert(), c.Require()
-
-	out, err := client.ListShipments(&easypost.ListShipmentsOptions{
-		Purchased:       easypost.BoolPtr(false),
-		IncludeChildren: easypost.BoolPtr(false),
-	})
-	require.NoError(err)
-	require.NotEmpty(out)
-	assert.True(len(out.Shipments) > 0)
+	assert := c.Assert()
+
+	shipment, _ := client.CreateShipment(c.fixture.OneCallBuyShipment())
+
+	shipmentWithNewLabel, _ := client.GetShipmentLabel(shipment.ID, "ZPL")
+
+	assert.NotNil(shipmentWithNewLabel.PostageLabel.LabelZPLURL)
 }
 
-func (c *ClientTests) TestShipmentSmartrates() {
+func (c *ClientTests) TestShipmentInsure() {
```

Comment:
Please ensure you are including all the comments from the other test suites:

```
      # If the shipment was purchased with a USPS rate, it must have had its insurance set to `0` when bought
      # so that USPS doesn't automatically insure it so we could manually insure it here.
```